roqoqo-qryd/src/tweezer_devices.rs

```diff
@@ -453,12 +465,20 @@ impl TweezerDevice {
         gate_time: f64,
         layout_name: Option<String>,
     ) -> Result<(), RoqoqoBackendError> {
-        self.qubit_to_tweezer = None;
+        if !ALLOWED_NATIVE_GATES.contains(&hqslang) {
+            return Err(RoqoqoBackendError::GenericError {
+                msg: format!(
+                    "Error setting the gate time of a single-qubit gate. Gate {} is not supported.",
```

Comment:
This doesn't seem to match the list of native gates, as some of them are non single-qubit gates